CDCgov/prime-reportstream
# Introduce IntelliJ formatting settings, update ktlint

This PR adds configs and IDE settings to the repo to establish consistent formatting settings for json/fhir, yml, and kotlin files. As part of this, it also updates ktlint to the latest version. The ktlint gradle plugin was updated in a recent dependabot update, so it didn't need to be included in this PR.

Some references that might be useful:

- [ktlint rule descriptions](https://pinterest.github.io/ktlint/1.0.0/rules/standard/)
- [kotlin code conventions](https://kotlinlang.org/docs/coding-conventions.html)

Test Steps:
1. Run usual test suite
2. Run `./gradlew ktlintCheck` and `./gradlew ktlintFormat` to ensure those work
3. Follow the steps at [prime-router/docs/getting-started/contributing.md](https://github.com/CDCgov/prime-reportstream/compare/master...platform/jimmy/11503-introduce-formatters#diff-171b35d34ae352bbe457ebac9586451090d8a09b7fc43bbbf28243b7b9199445) to set up the IDE formatter
4. Make some formatting changes, and verify that they are reformatted on save.
2. Run `./gradlew ktlintCheck` again to ensure the linter is still happy.

## Changes
This changeset is way too big to review everything. For the purposes of the review, focus on the following files:

- [.gitignore](https://github.com/CDCgov/prime-reportstream/compare/master...platform/jimmy/11503-introduce-formatters#diff-bc37d034bad564583790a46f19d807abfe519c5671395fd494d8cce506c42947): This was needed to track the IntelliJ code style settings in source control.
- [prime-router/.editorconfig](https://github.com/CDCgov/prime-reportstream/compare/master...platform/jimmy/11503-introduce-formatters#diff-686e7830a3adb458e3770e566a37af3ef30d59404db575aeac8f789b0a542f28): This covered some configurations for ktlint.
- [prime-router/.idea/codeStyles/Project.xml](https://github.com/CDCgov/prime-reportstream/compare/master...platform/jimmy/11503-introduce-formatters#diff-dcde54982a6d3943c7a9de27dc7923b3830183b2a77904e080208ff07acc25c2): This contains the formatting settings for the IntelliJ project.
- [prime-router/.idea/codeStyles/codeStyleConfig.xml](https://github.com/CDCgov/prime-reportstream/compare/master...platform/jimmy/11503-introduce-formatters#diff-fd5d69207d6846211bbe818937c25a2a0b331dfbb0fb37f642fb22d23f485fd3): This points to the Project.xml file for code style settings.
- [prime-router/build.gradle.kts](https://github.com/CDCgov/prime-reportstream/compare/master...platform/jimmy/11503-introduce-formatters#diff-068a6d1fb39db0c1087c194868d0180c7d7c48bad67e3c0473697a27bf2bdf4d): This updates the ktlint version.
- [prime-router/docs/getting-started/contributing.md](https://github.com/CDCgov/prime-reportstream/compare/master...platform/jimmy/11503-introduce-formatters#diff-171b35d34ae352bbe457ebac9586451090d8a09b7fc43bbbf28243b7b9199445): This covers documentation around setting up the automatic IDE formatting in IntelliJ.

The rest of the changes are automatic format changes either made by the IDE or by `ktlintFormat`

## Checklist

### Testing

## Linked Issues
- Addresses #11503

## To Be Done
Optionally, stories can be added to re-enable some of the disabled linting rules and make the necessary refactors to allow that to happen. There is no urgency to that, so I have opted not to create a story to track it, but I can if it would be useful.
